ui/src/main/java/com/fincity/saas/ui/document/URIPath.java

```diff
@@ -0,0 +1,140 @@
+import com.fincity.saas.commons.util.LogUtil;
+import com.fincity.saas.ui.enums.URIType;
+import com.fincity.saas.ui.model.KIRunFxDefinition;
+import com.fincity.saas.ui.model.RedirectionDefinition;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import lombok.experimental.Accessors;
+import reactor.core.publisher.Mono;
+import reactor.util.context.Context;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@Document(collection = "uri_path")
+@CompoundIndex(def = "{'appCode': 1, 'name': 1, 'clientCode': 1}", name = "uriFilteringIndex")
+@Accessors(chain = true)
+@NoArgsConstructor
+@ToString(callSuper = true)
+public class URIPath extends AbstractOverridableDTO<URIPath> {
+
+	@Serial
+	private static final long serialVersionUID = 2627066085463822531L;
+
+	private String pathString;
+	private URIType uriType;
+	private Boolean isAuthenticated;
+
+	private List<HttpMethod> httpMethods;
+	private List<String> headers;
+	private List<String> queryParams;
+	private List<String> whitelist;
+	private List<String> blacklist;
+
+	private KIRunFxDefinition kiRunFxDefinition;
+	private RedirectionDefinition redirectionDefinition;
+
+	public URIPath(URIPath uriPath) {
+
+		super(uriPath);
+
+		this.pathString = uriPath.pathString;
+		this.uriType = uriPath.uriType;
+		this.isAuthenticated = uriPath.isAuthenticated;
+		this.httpMethods = CloneUtil.cloneMapList(uriPath.httpMethods);
+		this.headers = CloneUtil.cloneMapList(uriPath.headers);
+		this.queryParams = CloneUtil.cloneMapList(uriPath.queryParams);
+		this.whitelist = CloneUtil.cloneMapList(uriPath.whitelist);
+		this.blacklist = CloneUtil.cloneMapList(uriPath.blacklist);
+		this.kiRunFxDefinition = CloneUtil.cloneObject(uriPath.kiRunFxDefinition);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public Mono<URIPath> applyOverride(URIPath base) {
+		if (base == null) {
+			return Mono.just(this);
+		}
+
+		return FlatMapUtil.flatMapMonoWithNull(
+				() -> DifferenceApplicator.apply(this.httpMethods, base.httpMethods),
+				hm -> DifferenceApplicator.apply(this.headers, base.headers),
+				(hm, h) -> DifferenceApplicator.apply(this.queryParams, base.queryParams),
+				(hm, h, qp) -> DifferenceApplicator.apply(this.whitelist, base.whitelist),
+				(hm, h, qp, w) -> DifferenceApplicator.apply(this.blacklist, base.blacklist),
+				(hm, h, qp, w, b) -> {
+					this.httpMethods = (List<HttpMethod>) hm;
+					this.headers = (List<String>) h;
+					this.queryParams = (List<String>) qp;
+					this.whitelist = (List<String>) w;
+					this.blacklist = (List<String>) b;
+
+					if (this.pathString == null)
+						this.pathString = base.pathString;
+					if (this.uriType == null)
+						this.uriType = base.uriType;
+					if (this.isAuthenticated == null)
+						this.isAuthenticated = base.isAuthenticated;
+					if (this.kiRunFxDefinition == null)
+						this.kiRunFxDefinition = CloneUtil.cloneObject(base.kiRunFxDefinition);
```

Comment:
Need to use difference applicator and extrator for both kirunfx def and redirection def objects.